src/api/app/models/workflow/step/branch_package_step.rb

```diff
@@ -8,48 +8,24 @@ def call(options = {})
     # FIXME: Support closed/merged/reopened PRs
     return if scm_webhook.closed_merged_pull_request? || scm_webhook.reopened_pull_request?
 
-    branched_package = find_or_create_branched_package
-
-    add_or_update_branch_request_file(package: branched_package)
-
     workflow_filters = options.fetch(:workflow_filters, {})
-    create_or_update_subscriptions(branched_package, workflow_filters)
 
-    workflow_repositories(target_project_name, workflow_filters).each do |repository|
-      # TODO: Fix n+1 queries
-      workflow_architectures(repository, workflow_filters).each do |architecture|
-        # We cannot report multibuild flavors here... so they will be missing from the initial report
-        SCMStatusReporter.new({ project: target_project_name, package: target_package_name, repository: repository.name, arch: architecture.name },
-                              scm_webhook.payload, @token.scm_token).call
-      end
+    # Updated PR
+    if scm_webhook.updated_pull_request? && target_package.present?
+      update_subscriptions(target_package, workflow_filters)
+    else # New PR or updated PR but target_package is not set
+      create_target_package
+      create_subscriptions(target_package, workflow_filters)
     end
 
-    branched_package
+    add_or_update_branch_request_file(package: target_package)
```

Comment:
what I don't understand @saraycp  is why this `add_or_update_branch_request_file` is different than `create_or_update_subscriptions`? Shouldn't we then break it in two methods?

Reply:
If we had two methods, they'll be identical. I think I'm going to rename the method, that's better

Reply:
done!